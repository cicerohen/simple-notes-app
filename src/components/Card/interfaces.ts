export interface CardPropsInterface {
  children?: JSX.Element | Array<JSX.Element>;
  className?: string;
}
