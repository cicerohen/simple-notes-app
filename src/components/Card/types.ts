import { FlattenSimpleInterpolation } from 'styled-components';

export type cardActiveMixinType = ({
  borderColor
}: {
  borderColor: string;
}) => FlattenSimpleInterpolation;

export interface CardPropsInterface {
  children?: JSX.Element | Array<JSX.Element>;
  className?: string;
}
