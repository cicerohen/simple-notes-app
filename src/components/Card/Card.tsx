import React, { FC } from 'react';
import styled from 'styled-components';

const Wrapper = styled.section`
  box-shadow: 0 0 5px #999;
`;

interface Props {
  children?: object | Array<object>;
}

const Card: FC<Props> = ({ children }: Props) => <Wrapper>{children}</Wrapper>;

export default Card;
