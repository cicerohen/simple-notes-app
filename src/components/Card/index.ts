export { default } from './Card';
export * from './interfaces';
