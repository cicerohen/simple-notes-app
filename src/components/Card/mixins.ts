import { css } from 'styled-components';
import { darken } from 'polished';

import { cardActiveMixinType } from './types';

export const cardActiveMixin: cardActiveMixinType = ({ borderColor }) => css`
  box-shadow: 0 0 2px ${darken('0.1', borderColor)};
`;
