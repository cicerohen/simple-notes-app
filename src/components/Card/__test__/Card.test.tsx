import React from 'react';
import { render } from '@testing-library/react';
import Card from '../Card';

describe('<Anchor /> Tests', () => {
  let component;

  beforeEach(() => {
    component = render(<Card />);
  });

  it('should render correctly without props', () => {
    expect(component.container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    component.rerender(
      <Card>
        <div>children</div>
      </Card>
    );
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
