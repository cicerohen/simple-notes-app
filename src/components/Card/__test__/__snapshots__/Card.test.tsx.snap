// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> Tests should render correctly with props 1`] = `
.c0 {
  box-shadow: 0 0 5px #999;
}

<section
  class="c0"
>
  <div>
    children
  </div>
</section>
`;

exports[`<Anchor /> Tests should render correctly without props 1`] = `
.c0 {
  box-shadow: 0 0 5px #999;
}

<section
  class="c0"
/>
`;
